### PR TITLE
Refresh progressive enhancement common concerns

### PR DESCRIPTION
**Pull request by hollsk**
@andrewmee wrote the original version of this back in 2016, and I've noticed that the formulation of the question, almost 10 years later, is now changing to "how many users turn JavaScript off", instead of being framed as an older technology support thing. 

So, since it's been a while, I've reframed his original work to fit the currently more common variation of the question, expanded on his points, and linked out to relevant pages. If we are still seeing the question as it was originally formulated, then please tell me and I'll work it back in!

### CODE REVIEW

**Review by andrewmee — APPROVED** [review]
Excellent reframing - much improved IMO

**Comment by hollsk** [comment]
You were always my favourite FED!

File: practices/progressive-enhancement.md [hunk 1]
```diff
@@ -42,15 +43,23 @@ With JavaScript we acknowledge that, given our [browser support chart](graded-br
...
+
+Progressive enhancement is not just a solution for clients who have deliberately deprived themselves of JavaScript, it's also a failsafe for EVERY client which guarantees your site/application is as useful as it can be to everyone all the time.
+
+### OK, so how often does JavaScript fail? How can we test for that?
+
+Usually this question comes from the Extreme Programming thinking of "[You Ain't Gonna Need It](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)" - the desire to only implement features when they're necessary. However, as the previous section ("How many users turn JavaScript off anyway?") makes it clear, JavaScript delivery failure happens to everyone, some of the time. Robustness is a core requirement for our websites and applications, not an optional feature. Stuart Langridge illustrates why you need to think in terms of JavaScript failing for a % of _visits_, not _users_.
```
Comment (jon-stevens) [inline practices/progressive-enhancement.md:54]:
```suggestion
Usually this question comes from the Extreme Programming thinking of "[You Ain't Gonna Need It](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)" - the desire to only implement features when they're necessary. However, as the previous section ("How many users turn JavaScript off anyway?") makes clear, JavaScript delivery failure happens to everyone, some of the time. Robustness is a core requirement for our websites and applications, not an optional feature. Stuart Langridge illustrates why you need to think in terms of JavaScript failing for a % of _visits_, not _users_.
```

**Comment by jpw** [comment]
Wonderful stuff, thanks very much everyone. LGTM.

**Pull request merged**